docs/architecture.md

```diff
@@ -0,0 +1,292 @@
+# All About `armada-spark`
+
+## Running `armada-spark`
+
```

Comment:
This section looks good, but probably needs an addtional subsection on running python jobs.  Check the latest submit script in master.

Reply:
Addressed in 9847f2c